core/src/main/java/hudson/tasks/CommandInterpreter.java

```diff
@@ -128,8 +127,7 @@ public boolean perform(AbstractBuild<?,?> build, Launcher launcher, TaskListener
                 // on Windows environment variables are converted to all upper case,
                 // but no such conversions are done on Unix, so to make this cross-platform,
                 // convert variables to all upper cases.
-                for(Map.Entry<String,String> e : build.getBuildVariables().entrySet())
-                    envVars.put(e.getKey(),e.getValue());
+                envVars.putAll(build.getBuildVariables());
```

Comment:
See https://github.com/jenkinsci/jenkins/pull/6046#discussion_r767137478

Reply:
Thanks, I'll remove the commit.